README.md

```diff
@@ -82,7 +84,210 @@ Requirements:
        ```
 
 Build:
-------
+-----------------
+  Aladdin currently supports two ways to build.
+
+  1. Configure with CMake, then build with Make :
+
+       If you are familiar with CMake, or you are interested in a script
+       which can build out-of-source and offer automatical LLVM/Clang
+       installation, choose [CMake](#build-with-cmake)
+
+  2. Directly build with Makefile :
+
+       If you want to keep all the things simple, or you do not have
+       CMake, choose [Makefile](#build-with-makefile)
+
+
+Build with CMake:
+-----------------
+CMake is a configure tool which allows you to out-of-source build.
+Aladdin Requires CMake newer than 2.8.12. By default, CMake
+searches for LLVM 3.4.
+
+
+1. Set `$BOOST_ROOT`,  `TRACER_HOME`, `LLVM_HOME` to let Aladdin know where the tools and libraries are.
```

Comment:
Does this assume users have already installed LLVM? If LLVM_HOME here where their LLVM is installed or where CMake is gonna install LLVM for them? Should we clarify here?